.github/actions/generate-commit-list-v1/src/getParsedCommitList.ts

```diff
@@ -0,0 +1,37 @@
+import getCommitType from './getCommitType'
+import { GetParsedCommitListParams, ParsedCommitList } from './types'
+
+/**
+ * Parses the raw commit list into an array of commit objects.
+ * @param {object} GetParsedCommitListParams Raw commit list and repository URL.
+ * @returns An array of parsed commits.
+ */
+
+export default function getParsedCommitList({
+  rawCommitList,
+  repository
+}: GetParsedCommitListParams): Array<ParsedCommitList> {
+  const parsedCommits: Array<ParsedCommitList> = []
+
+  for (const commit of rawCommitList) {
+    // group sha and title together
+    //eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const shaAndTitle = commit.match(/^(.{7}) (.+?)(?:\s\(#\d+\))?$/)!
+    const sha = shaAndTitle[1]
+    const title = shaAndTitle[2]!
+    const type = getCommitType(title)
+    const id = commit.match(/#(\d+)/)?.[0].replace('#', '') || null
```

Comment:
We'll want to make sure this is only grabbing the id from the end and not any random `#number` from the title. Also since you're using a group for the numbers you can use that and not have to replace the `#`

```suggestion
    const id = commit.match(/\(#(\d+)\)$/)?.[1] || null
```